civis/run_joblib_func.py

```diff
@@ -0,0 +1,81 @@
+"""
+This is an executable intended for use with a joblib backend
+for the Civis platform. It takes a Civis File ID representing
+a joblib-serialized callable as an argument, downloads the file,
+deserializes it, calls the callable, serializes the result,
+and uploads the result to another Civis File. The output file's ID
+will be set as an output on this run.
+"""
+from __future__ import absolute_import, print_function
+
+from datetime import datetime, timedelta
+from io import BytesIO
+import os
+import sys
+
+import civis
+import joblib
+from joblib.my_exceptions import TransportableException
+from joblib.format_stack import format_exc
+
+
+def worker_func(func_file_id):
+    # Have the output File expire in 7 days.
+    expires_at = (datetime.now() + timedelta(days=7)).isoformat()
+
+    client = civis.APIClient()
+    job_id = os.environ.get('CIVIS_JOB_ID')
+    run_id = os.environ.get('CIVIS_RUN_ID')
+    if not job_id or not run_id:
+        raise RuntimeError("This function must be run inside a "
+                           "Civis container job.")
+
+    func_buffer = BytesIO()
+    civis.io.civis_to_file(func_file_id, func_buffer)
```

Comment:
Note that even for functions being executed here which are not idempotent, the downloading and uploading of result steps are idempotent.